testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/response/PublisherResponseTest.java

```diff
@@ -95,27 +107,33 @@ public void testText() throws Exception
     * @tpTestDetails Resource method returns Publisher<String>, throws exception immediately.
     * @tpSince RESTEasy 4.0
     */
+   @SuppressWarnings("unchecked")
    @Test
    public void testTextErrorImmediate() throws Exception
    {
-      Invocation.Builder request = client.target(generateURL("/text-error-immediate")).request();
-      Response response = request.get();
-      String entity = response.readEntity(String.class);
-      Assert.assertEquals(200, response.getStatus());
-      Assert.assertEquals("Got it", entity);
-
+      FlowableRxInvoker invoker = client.target(generateURL("/text-error-immediate")).request().rx(FlowableRxInvoker.class);
+      Flowable<String> flowable = (Flowable<String>) invoker.get();
+      AtomicReference<Object> value = new AtomicReference<Object>();
+      flowable.subscribe(
+            (String s) -> {},
+            (Throwable t) -> {value.set(t);latch.countDown();},
+            () -> {});
+      latch.await();
+      ClientErrorException cee = (ClientErrorException)value.get();
+      Assert.assertEquals("Got it", cee.getResponse().readEntity(String.class));
+      
       // make sure the completion callback was called with with an error
-      request = client.target(generateURL("/callback-called-with-error")).request();
-      response = request.get();
+      Builder request = client.target(generateURL("/callback-called-with-error")).request();
+      Response response = request.get();
       Assert.assertEquals(200, response.getStatus());
       response.close();
    }
-
    /**
     * @tpTestDetails Resource method returns Publisher<String>, throws exception in stream.
     * @tpSince RESTEasy 4.0
     */
    @Test
+   @Ignore // Doesn't currently work.
```

Comment:
Is this a regression?

Reply:
@marekkopecky, this is a new version of an existing test, which is now PublisherResponseNoStreamTest.testTextErrorDeferred(). The original still works, so it's not a regression. I've added a comment.